### PR TITLE
DX: re-apply PHP CS Fixer, partially

### PR DESCRIPTION
**Pull request by keradus**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Issues        | Fix CS 
| License       | MIT

One further step to clean the CS status-quo, aiming for all-green in long-term.
~90 files remaining, yet they may require some further update on tool itself

best to review commit by commit

### CODE REVIEW

File: src/Symfony/Component/Uid/BinaryUtil.php [hunk 1]
```diff
@@ -128,7 +128,7 @@ public static function hexToDateTime(string $time): \DateTimeImmutable
 
             if (self::TIME_OFFSET_BIN <= $time) {
                 $time = self::add($time, self::TIME_OFFSET_COM2);
-                $time[0] = $time[0] & "\x7F";
+                $time[0] &= "\x7F";
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Uid/BinaryUtil.php]: AFAIK, this is not the same for string indexes. Would need double checking if this was fixed in recent versions of PHP.

Comment (keradus) [reply src/Symfony/Component/Uid/BinaryUtil.php]: What is the string index here? (not the `[0]`, right?)

It is causing CI to fail, let's revert for now indeed

Comment (nicolas-grekas) [reply src/Symfony/Component/Uid/BinaryUtil.php]: yes it is, same for the other similar changes in the PR

Comment (keradus) [reply src/Symfony/Component/Uid/BinaryUtil.php]: > this is not the same for string indexes.

can you elaborate on that `string index` a bit? I'm not aware what is the underlying issue here.

> same for the other similar changes in the PR

after revert ot `&=` , CI is green (ok, except of fabbot for file license-headers)

Comment (nicolas-grekas) [reply src/Symfony/Component/Uid/BinaryUtil.php]: Sure: https://3v4l.org/GI1NW

**Comment by nicolas-grekas** [comment]
Thank you @keradus.

File: src/Symfony/Component/Uid/BinaryUtil.php (hunk 1 above)
Comment (keradus) [reply src/Symfony/Component/Uid/BinaryUtil.php]: Rule marked as risky, at least: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7418
perhaps, with time we can have better detection if we touch index or not

**Pull request merged**